src/components/TransactionBlock/KlimaStakingTransactionBlock.tsx

```diff
@@ -279,10 +280,39 @@ const KlimaStakingTransactionBlock = ({
           return;
         }
 
+        const sdkChain = getSdkForChainId(CHAIN_ID.POLYGON);
+        const gasInfo = await sdkChain?.getGatewayGasInfo();
+
+        let priceUsd = await getAssetPriceInUsd(CHAIN_ID.POLYGON,ethers.constants.AddressZero,sdk);
+        if (!gasInfo || !priceUsd) {
+          setTransactionBlockFieldValidationError(transactionBlockId, 'amount', `No Offer found`);
+          resetRoutes();
+          return;
+        }
+
+        let estimatedGas: BigNumberish = 850000;
+        let currentGasPrice = gasInfo?.fast;
```

Comment:
We have a nullable here (`?.`) - what happens when `currentGasPrice` is null on this line?

`let gasFees = currentGasPrice.mul(estimatedGas);`

Reply:
Yes. We don't need this **"?"**, as we already validate in the above check.